docs/rfcs/002-branch-management.md

```diff
@@ -0,0 +1,95 @@
+# Branch Management From Command Palette
+
+## Status
+
+Proposed
+
+## Summary
+
+Efficient branch management that is in line with existing Atom conventions.
+
+## Motivation
+
+Current branch management is incomplete (e.g. no deleting of branches) and not consistent with other conventions found in the Atom UI for comparable actions. It is hard to switch or create new branches with just keyboard input.
+
+## Explanation
+
+The new branch selector exists largely in the [command palette](https://github.com/atom/command-palette). The only UI element is the pre-existing `branch-view` tile in the status bar:
+
+![existing branch view tile](https://cldup.com/CE_61W1uBk.png)
+
+### Command Palette
+
+The command palette contains the following new commands:
+- Create new branch (`github:create-new-branch`)
+- Checkout branch (`github:checkout-branch`)
+- Delete branch (`github:delete-branch`)
+- Rename branch (`github:rename-branch`)
+
+#### `github:create-new-branch`
+
+Executing `github:create-new-branch` opens a text input dialog, similar to when creating a new file asks for the filename. When pressing "Escape", the dialog closes without creating a branch, when pressing "Return" a branch of that name is created and switched to.
+
+![Branch name input](https://cldup.com/GhKgFE_zTy.png)
+
+#### `github:checkout-branch`
+
+Executing `github:checkout-branch` opens a filterable command palette list of local branches, similar to the list of available grammars when executing `grammar-selector:show`. Selecting a branch from that list switches to that branch.
```

Comment:
How would you feel about including remote tracking branches in the filter list? We could style or decorate them slightly differently as a visual cue, and then create a new local branch that tracks them like `git checkout -b` does. Might be a nice way to fill in some missing functionality from the existing branch selector.